tasks/basecalling/task_dorado_basecall.wdl

```diff
@@ -0,0 +1,54 @@
+version 1.0
+
+task basecall {
+  input {
+    File input_file                    # Single POD5 file for scatter processing
+    String dorado_model = "sup"         # Default model to 'sup', can be overridden with full model name see docs
+    String kit_name                    # Sequencing kit name
+    Int disk_size = 100
+    Int memory = 32
+    Int cpu = 8
+    String docker = "us-docker.pkg.dev/general-theiagen/staphb/dorado:0.8.0"
+  }
+
+  command <<< 
+    set -euo pipefail
+
+    # Create output directory for SAM files
+    sam_output="output/sam/"
+    mkdir -p "$sam_output"
+
+    echo "### Starting basecalling for ~{input_file} ###"
+
+    base_name=$(basename "~{input_file}" .pod5)
+    sam_file="$sam_output/${base_name}.sam"
+
+    echo "Processing ~{input_file}, output: $sam_file"
+
+    # Run Dorado basecaller
+    dorado basecaller \
+      "~{dorado_model}" \
+      "~{input_file}" \
+      --kit-name ~{kit_name} \
+      --emit-sam \
+      --no-trim \
+      --output-dir "$sam_output" \
+      --verbose || { echo "ERROR: Dorado basecaller failed for ~{input_file}"; exit 1; }
+
```

Comment:
consider re-naming output SAM files to avoid duplicate SAM filenames produced across different shards

Reply:
perhaps `basename ~{input_file}` so final resulting file is the `basename.sam`

Reply:
I think he's doing this see line 23

Reply:
wait, i think that needs to be updated so line 35 references $sam_file instead

Reply:
Fraser and I chatted about this and believe we have a solution. Dorado allows you to name the output directory, but not the specific output SAM file (AFAIK)

so I think he's going to add some code to rename the output SAM file based on the input POD5 filename which should be a unique string instead of `$sam_output/call_<datetime-stamp>.sam` which is where I found duplicate filenames because of the very slim chance that dorado ran at the exact same time on 2 separate input pod5 files :/